### PR TITLE
[BEAM-1676] Internalize ClassPath from Guava 21.0.

### PR DESCRIPTION
**Pull request by staslev**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by staslev** [comment]
R: @dhalperi

**Comment by dhalperi** [comment]
Rather than lumping it inside ApiSurfaceTest, could we put it a package-private ClassPath.java (e.g.), so that `ApiSurfaceTest` is essentially a no-op both here and once we can delete this code? Seems safer going forward.

**Comment by staslev** [comment]
You mean the `ApiSurface` class right?
Nice, I'll get it done.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/util/ApiSurface.java [hunk 1]
```diff
@@ -825,9 +825,9 @@ private boolean exposed(int modifiers) {
    */
   public static ApiSurface getSdkApiSurface(final ClassLoader classLoader) throws IOException {
     return ApiSurface.ofPackage("org.apache.beam", classLoader)
-        .pruningPattern("org[.]apache[.]beam[.].*Test")
-        // Exposes Guava, but not intended for users
-        .pruningClassName("org.apache.beam.sdk.util.common.ReflectHelpers")
-        .pruningPrefix("java");
+                     .pruningPattern("org[.]apache[.]beam[.].*Test")
```
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/util/ApiSurface.java]: I would prefer to not change the indentation scheme "incidentally", but up to you.

**Comment by dhalperi** [comment]
LGTM once green. Thanks for this!!

**Pull request closed without merging**